.github/workflows/run_tests.yml

```diff
@@ -63,32 +84,65 @@ jobs:
             ${{ steps.vcpkg-cache.outputs.archives_dir }}
             ${{ steps.vcpkg-cache.outputs.downloads_dir }}
           key: vcpkg
+
+      - name: Set up ccache for liboqs
+        uses: hendrikmuhs/ccache-action@v1.2
+        if: ${{ runner.os != 'macOS' }}
+        with:
+          variant: sccache
+          key: liboqs-ccache-${{ matrix.os }}
+
       - name: Install Linux dependencies
         if: ${{ runner.os == 'Linux' }}
-        run: sudo apt install -y --no-install-recommends libnettle8 libsodium-dev libssl-dev libkrb5-dev ssh
+        run: sudo apt install -y --no-install-recommends libnettle8 libsodium-dev libssl-dev libkrb5-dev ssh cmake ninja-build
+
       - name: Install macOS dependencies
         if: ${{ runner.os == 'macOS' }}
         run: brew install nettle liboqs libsodium openssl
+
       - name: Provide OpenSSL 3
         if: ${{ runner.os == 'macOS' && matrix.openssl-version == '3' }}
         run: echo "/usr/local/opt/openssl@3/bin" >> $GITHUB_PATH
+
       - name: Install Windows dependencies
         if: ${{ runner.os == 'Windows' }}
         run: vcpkg install libsodium nettle openssl --triplet x64-windows
+
+      - name: Install liboqs (Linux)
+        if: ${{ runner.os == 'Linux' }}
+        working-directory: liboqs
+        run: |
+          cmake -GNinja -Bbuild . -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_SHARED_LIBS=ON -DOQS_BUILD_ONLY_LIB=ON -DCMAKE_C_COMPILER_LAUNCHER=sccache
+          cmake --build build
+          sudo cmake --install build
+
+      - name: Install liboqs (Windows)
+        if: ${{ runner.os == 'Windows' }}
+        shell: pwsh
+        working-directory: liboqs
+        run: |
+          cmake -GNinja -Bbuild . -DBUILD_SHARED_LIBS=ON -DOQS_BUILD_ONLY_LIB=ON -DCMAKE_C_COMPILER_LAUNCHER=sccache
+          cmake --build build
+          copy build\bin\liboqs.dll build\bin\oqs.dll
```

Comment:
For some reason, the built DLL is called `liboqs.dll` which cannot be found by Python with `find_library("oqs")`.

Reply:
I've had trouble with this in the past. For nettle, I ended up with the following in `crypto/umac.py`.

```python
_nettle_lib = 'libnettle-6' if sys.platform == 'win32' \
                            else ctypes.util.find_library('nettle')

if _nettle_lib: # pragma: no branch
    _nettle = ctypes.cdll.LoadLibrary(_nettle_lib)
```

In this case, the DLL was libnettle-6.dll, installed in the Python version's main directory (`C:\Program Files\Python310` on my system). I think if you make a similar win32 check and skip find_library, something similar to the above should hopefully work with liboqs.

Since I haven't tried building liboqs on Windows yet, I didn't make the same change there yet.

Reply:
I think I figured out why. If CMake uses the GNU toolchain it will add the `lib` prefix to the DLL. I've included a third-party action that initializes an MSVC environment so that CMake will use the MSVC toolchain.

Reply:
Not sure if it's the same reason for nettle, though. It seems to be a lot trickier when vcpkg is involved.

Reply:
So find_library() works, but only when the filename is oqs.dll, not liboqs.dll? I'm not a big windows user, but that wouldn't surprise me. What's strange, though, is that other third-party packages I use seem to want the "lib" version. For instance, pynacl seems to look for "libsodium-<version>":

```python
__SONAMES = (23, 18, 17, 13, 10, 5, 4)

def _get_nacl():
    '''
    Locate the nacl c libs to use
    '''
    if sys.platform.startswith('win'):
        try:
            return ctypes.cdll.LoadLibrary('libsodium')
        except OSError:
            pass
        for soname_ver in __SONAMES:
            try:
                return ctypes.cdll.LoadLibrary(
                    'libsodium-{0}'.format(soname_ver)
                )
            except OSError:
                pass
        msg = 'Could not locate nacl lib, searched for libsodium'
        raise OSError(msg)
```

For Nettle. the binaries I found online has a similar naming convention (libnettle-6.dll), but interestingly it seems that at least on win64 they actually went with the bane 'hogweed.dll' without a prefix/suffix for another library nettle depended on!

Knowing that there is a way to make this work with find_library() on Windows, perhaps I could look into another approach where I try find_library() first, and only when it doesn't find it do I have it fall back to the "lib" version of the name. For instance:

```python
_nettle_lib = ctypes.util.find_library('nettle') or 'libnettle-6'
```

Actually, playing around a bit, I can actually pass in something like 'libnettle-6' directly to find_library() and it'll return a full path for the DLL, so I could also just have it try 'nettle', 'libnettle', and 'libnettle-6' until it found one that works.

For the sake of your check-in here, I'm good with just leaving it as the existing find_library() call if you have that working. I'll think about f I want to allow for the "lib" version of the name as a fallback, but that can be done as a separate checkin.

Thanks for your work on this! Let me know when you've got something you're comfortable merging.

Reply:
> So find_library() works, but only when the filename is oqs.dll, not liboqs.dll? I'm not a big windows user, but that wouldn't surprise me.

Yes, it seems to be the case. On Linux/Unix `find_library` expects the library name without `lib`, on Windows, it is just the name of the DLL.

>What's strange, though, is that other third-party packages I use seem to want the "lib" version. For instance, pynacl seems to look for "libsodium-":
> ...
> 
> For Nettle. the binaries I found online has a similar naming convention (libnettle-6.dll), but interestingly it seems that at least on win64 they actually went with the bane 'hogweed.dll' without a prefix/suffix for another library nettle depended on!
> 

My understanding is that the naming of the DLL depends on the build system, which usually makes its decision based on the discovered toolchain. In both CMake and Meson, DLLs have the `lib` prefix when they are built with the MinGW toolchain, but not with MSVC. Specifically, CMake automatically sets `CMAKE_SHARED_LIBRARY_PREFIX` to `lib` if the toolchain is GNU (MinGW), or keeps it empty if it detects MSVC.

`hogweed.dll` remains a mystery for me - it is a part of Nettle, and their Autotools build system doesn't seem to support MSVC, and in its [configure.ac](https://github.com/gnutls/nettle/blob/master/configure.ac) it refers to the library as `libhogweed.{so,dylib,a,dll,...}`, so I'm not sure how that file is produced...

> Knowing that there is a way to make this work with find_library() on Windows, perhaps I could look into another approach where I try find_library() first, and only when it doesn't find it do I have it fall back to the "lib" version of the name. For instance:
> 
> ```python
> _nettle_lib = ctypes.util.find_library('nettle') or 'libnettle-6'
> ```
> 
> Actually, playing around a bit, I can actually pass in something like 'libnettle-6' directly to find_library() and it'll return a full path for the DLL, so I could also just have it try 'nettle', 'libnettle', and 'libnettle-6' until it found one that works.
> 

That sounds like a good idea!

Reply:
Ok - I've gone ahead and made the change mentioned above around load_library() for Nettle and OQS in commit f1f36ce. For OQS in particular, it should support either "oqs.dll" or "liboqs.dll" on Windows.

I'll be merging this PR shortly - I really appreciate your help on this!

Reply:
No problem, happy to help!